### PR TITLE
l10n: zh_CN: Update Git Glossary: "commit message"

### PR DESCRIPTION
**Pull request by oldsharp**
Add "commit message" to glossary.  Translate it as "提交信息".

Signed-off-by: Ray Chen oldsharp@gmail.com

### CODE REVIEW

**Comment by louy2** [comment]
提议：commit 改译作「保证」
理由：
1. commit 字典义有「保证」，没有「提交」
2. commit 功能是「保证能使仓库回到该状态」，不只是「提交」
3. commit 动词名词同形，「保证」也是，「提交」不是

**Comment by oldsharp** [comment]
On Tue, 25 Aug 2015, Yufan Lou wrote:

> 提议：commit 改译作「保证」
> 理由：
> 1. commit 字典义有「保证」，没有「提交」

Consulted the "Oxford Dictionary of English" for the "commit" entry
( http://www.oxforddictionaries.com/definition/english/commit ):
1. (commit something to) Transfer something to (a state or place
    where it can be kept or preserved)

And also the "Oxford Chinese Dictionary" (one of my iPhone's built-in
dictionaries):

  ④ (transfer, record) 记下
    ▸ to commit [something] to paper/writing
      将某事记在纸上/写下来
    ▸ to commit [something] to memory
      记住某事物

Both may refer to as "提交".

> 1. commit 功能是「保证能使仓库回到该状态」，不只是「提交」

As you said, `git-commit` may ensure a particular state of one's
repository - that's right, but this is based on the snapshot recording
operation.

1) From Git Manual (`man git-commit`):

   NAME
          git-commit - Record changes to the repository

   ...

   DESCRIPTION
          Stores the current contents of the index in a new commit
          along with a log message from the user describing the
          changes.

   ...

2) What `git-commit` actually do (`man git-commit-tree`):

   NAME
          git-commit-tree - Create a new commit object

3) As a common VCS term, "提交" may stand for "commit" or "check in".
   Both make sense, in my opinion.

> 1. commit 动词名词同形，「保证」也是，「提交」不是

Ah, I think "提交" can be a noun as well as a verb (at least in Git
context):
- 提交现有更改到仓库。(as a verb)
- 请回退到上一次提交。(as a noun)

> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/jiangxin/git/pull/10#issuecomment-134633309

What `git-commit` do is nothing more than 1) generate a snapshot of the
current index 2) record it to the core base.
So I think "commit" translated as "提交" is fair enough 😉

**Comment by louy2** [comment]
「记录」 may be even better. What I don't like about 「提交」 is its implying a client-server relationship. I'd like the name to be local in nature which is what distinguishes git from older VCS. 「记录」 also has all the above.

**Comment by oldsharp** [comment]
On Tue, 25 Aug 2015, Yufan Lou wrote:

> 「记录」 may be even better. What I don't like about 「提交」 is its
> implying a client-server relationship.

So you are worrying about someone may be confused with "commit" and
"submit", if we translate "commit" as "提交"?

Okay, here we should first have some concepts clarified.

The concept and term "commit" in revision control is _directly_
diverged from the data management in computer science - a commit is the
making of a set of tentative changes permanent.  Think about the
COMMIT, BEGIN WORK, ROLLBACK statements in transaction processing
within a RDBMS.

Now for the translation "提交".  In fact, what I think the origin of
translating "commit" as "提交" is the usage of "commit" in database
transaction: "atomic commit(原子提交)",
"two-phase commit protocol(两阶段提交协议)", etc.

These classic translations are way too common to be possibly confused
with "submit", since they have nothing to do with "submitting a form"
or something like that, nor does any client-server relationships
involved.

> I'd like the name to be local in nature which is what distinguishes
> git from older VCS.

Agreed.  That's a strong argument.  This is the reason why Git use
"commit" instead of "check-in".

> 「记录」 also has all the above.

Yeah, I agreed that "记录" is a good candidate.

However, there are still a few questions:
- How to translate the word "record"?  This word takes a great chance
  to appear together with "commit" in Git related documents.
- We have a strong reason and tradition to translate "commit" as
  "提交", as I described above.  What's the plan to solve that?

**Comment by louy2** [comment]
> git-commit - Record changes to the repository

git-记录 - 记录仓库的改动

"commit" is defined by "record", so I'm fine with using the same translation. It still reads natural IMO. If there happens to be such text as "record a commit", I'd rephrase it as "进行一次记录“。

I did not know about "Atomic commit" or 2PC before you mentioned, so I think it is not without doubt to assume git users know about such tradition and can automatically understand 提交 in RDBMS sense. (Thank you for raising it though, found some good reads about them.) I'd even try to replace those translations with 记录, if possible, since I regard it more accurate. I personally only refer commits by English when I'm discussing, so I am not sure whether "commit -> 提交" is already a tradition among Chinese dev.

**Comment by jiangxin** [comment]
Thanks all.
